extensions/builder.md

```diff
@@ -0,0 +1,180 @@
+# Builder Specification <!-- omit in toc -->
+
+This document specified the artifact format for a builder.
+
+A builder is an OCI image that provides a distributable build environment.
+
+A [platform][platform-spec] supporting the builder extension specification SHOULD allow users to configure the build environment with a provided builder.
+
+## Table of Contents <!-- omit in toc -->
+- [General Requirements](#general-requirements)
+  - [Builder API Version](#builder-api-version)
+  - [File/Directories](#filedirectories)
+  - [Environment Variables](#environment-variables)
+  - [Labels](#labels)
+- [Data Format](#data-format)
+  - [Labels](#labels-1)
+    - [`io.buildpacks.builder.metadata` (JSON)](#iobuildpacksbuildermetadata-json)
+    - [`io.buildpacks.buildpack.order` (JSON)](#iobuildpacksbuildpackorder-json)
+    - [`io.buildpacks.buildpack.layers` (JSON)](#iobuildpacksbuildpacklayers-json)
+    - [`io.buildpacks.lifecycle.apis` (JSON)](#iobuildpackslifecycleapis-json)
+
+## General Requirements
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+### Builder API Version
+This document specifies Builder API version `0.1`.
+
+Builder API versions:
+- MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
+- When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
+
+### File/Directories
+A builder MUST have the following directories/files:
+- `/cnb/order.toml` &rarr; As defined in the [platform specification][order-toml-spec]
+- `/cnb/stack.toml` &rarr; As defined in the [platform specification][stack-toml-spec]
+- `/cnb/lifecycle/<lifecycle binaries>` &rarr; An implementation of the lifecycle, which contains the required lifecycle binaries for [building images][lifecycle-for-build].
+
+In addition, every buildpack blob contained on a builder MUST be stored at the following file path:
+- `<CNB_BUILDPACKS_DIR>/<buildpack ID>/<buildpack version>/`
+
+If the buildpack ID contains a `/`, it MUST be replaced with `_` in the directory name.
+
+All specified files and directories are writeable by the build environment's User. 
```

Comment:
Do we want buildpacks to have permission to change the `order.toml` or the code of other buildpacks? The files above are all currently root owned in builders created by `pack`.

We do want the layers dir and app dir to be writable by the build user however and should probably call that out explicitly.

Reply:
I would assume that it definitely shouldn't be able to change the code of other buildpacks (except perhaps stackpacks). 

So should I change this to something like:
```
The `CNB_APP_DIR` and `CNB_LAYERS_DIR` are writeable by the build environment's User.
```
?

Reply:
Made a suggestion...